src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -457,11 +474,16 @@ private void ScanFormatSpecifier()
                         {
                             TrySetUnrecoverableError(_lexer.MakeError(pos, 1, ErrorCode.ERR_EscapedCurly, ch));
                         }
+                        else if (ch is '"' or ':')
+                        {
+                            TrySetUnrecoverableError(_lexer.MakeError(pos, 1, ErrorCode.ERR_UnexpectedCharacter, ch));
+                        }
```

Comment:
why was this change made?  it doesn't actually seem to match anything.

Reply:
i think we should make the minimal change necessary to match the runtime.

Reply:
it matches for \u0022 and \u003A